pokedex/profiles.lua

```diff
@@ -219,9 +215,67 @@ local function convert_to_rolling_profile_slot()
 	profiles.last_used = nil
 end
 
+-- HACKY, see caller
+local species_variant_cb
+function M.register_species_variant_cb(cb)
+	species_variant_cb = cb
+end
+
+local function upgrade_profiles()
+	local version = profiles and profiles.version or 1
+
+	local needs_upgrade = version ~= LATEST_PROFILE_VERSION
+	
+	if needs_upgrade then
+		for i=version,LATEST_PROFILE_VERSION-1 do
+			if false then
+
+				-- NOTE: If a new data upgrade is needed, update the above LATEST_PROFILE_VERSION value and add a new block here like so:
+				--elseif i == ??? then
+
+			elseif i == 2 then
+
+				if profiles.slots then
+
+					for s=1,#profiles.slots do
+						local old_party = profiles.slots[s].party
```

Comment:
This is so the profile screen doesn't show MissingNo images if you happened to have a pokemon with variants in your party.